tests/integration/test_charm.py

```diff
@@ -0,0 +1,269 @@
+            credentials[MYSQL]["password"],
+            credentials[MYSQL]["endpoints"],
+            credentials[MYSQL]["version"],
+            credentials[MYSQL]["read-only-endpoints"],
+        )
+
+    with MysqlConnector(config) as cursor:
+
+        rows = read_data_mysql(cursor, credentials[MYSQL]["username"])
+        # check that values are written in the table
+        check_my_sql_data(rows, credentials)
+
+    #  remove relation and test connection again
+    await ops_test.model.applications[DATA_INTEGRATOR].remove_relation(
+        f"{DATA_INTEGRATOR}:mysql", f"{MYSQL}:database"
+    )
+
+    await ops_test.model.wait_for_idle(apps=[MYSQL, DATA_INTEGRATOR])
+    await ops_test.model.add_relation(DATA_INTEGRATOR, MYSQL)
+    await ops_test.model.wait_for_idle(apps=[DATA_INTEGRATOR, MYSQL])
+
+    # join with another relation and check the accessibility of the previously created database
+    new_credentials = await fetch_action_get_credentials(
+        ops_test.model.applications[DATA_INTEGRATOR].units[0]
+    )
+
+    new_config = {
+        "user": new_credentials[MYSQL]["username"],
+        "password": new_credentials[MYSQL]["password"],
+        "host": new_credentials[MYSQL]["endpoints"].split(":")[0],
+        "database": DATABASE_NAME,
+        "raise_on_warnings": False,
+    }
+    # test connection with new credentials and check the previously committed data are present.
+    with MysqlConnector(new_config) as cursor:
+        rows = read_data_mysql(cursor, credentials[MYSQL]["username"])
+        # check that values are written in the table
+        check_my_sql_data(rows, credentials)
+
+
+@pytest.mark.skip  # skipping as we can't reconnect to same database (https://github.com/canonical/postgresql-k8s-operator/issues/64)
+async def test_deploy_and_relate_postgresql(ops_test: OpsTest):
+    """Test the relation with PostgreSQL and database accessibility."""
+    await asyncio.gather(
+        ops_test.model.deploy(
+            "postgresql", channel="edge", application_name=POSTGRESQL, num_units=1, series="focal"
+        )
+    )
+    await ops_test.model.wait_for_idle(apps=[POSTGRESQL])
+    assert ops_test.model.applications[POSTGRESQL].status == "active"
+    await ops_test.model.add_relation(DATA_INTEGRATOR, POSTGRESQL)
+    await ops_test.model.wait_for_idle(apps=[DATA_INTEGRATOR, POSTGRESQL])
+    assert ops_test.model.applications[DATA_INTEGRATOR].status == "active"
+
+    # get credential for PostgreSQL
+    credentials = await fetch_action_get_credentials(
+        ops_test.model.applications[DATA_INTEGRATOR].units[0]
+    )
+    # Build the complete connection string to connect to the database.
+    connection_string = build_postgresql_connection_string(credentials)
+    version = credentials[POSTGRESQL]["version"]
+
+    # test connection for PostgreSQL with retrieved credentials
+    with psycopg2.connect(connection_string) as connection, connection.cursor() as cursor:
```

Comment:
Done

Reply:
Uhm, psycopg2 already provides a context manager. I could see the reason for this if we had abstracted things out and we wanted to have a context with the same signature, in order to have some general code:

```
with Context(config) as cursor:
    create_table(cursor)
    insert_data(cursor)
```

and this could either be used for mysql, postgresql and mongo. But we are a long way from that and, as I said in the general comment, I would feel it is really not needed here, and would possibly be rather overengineered. 

Thus, I suppose we can also keep the psycopg2 context manager